assembly/src/todo/mod.rs

```diff
@@ -0,0 +1,432 @@
+use core::borrow::Borrow;
+
+use vm_core::{code_blocks::CodeBlock, CodeBlockTable, Felt, FieldElement, Operation, Program};
+
+use crate::{
+    parsers::{self, Instruction, Node},
+    procedures::Procedure,
+    ModuleProvider, ProcedureAst, ProcedureId, ProgramAst,
+};
+
+pub struct Assembler {
+    module_provider: Box<dyn ModuleProvider>,
+    cb_table: CodeBlockTable,
+    locals: Vec<Procedure>,
+    blocks: Vec<Vec<CodeBlock>>,
+    spans: Vec<Vec<Operation>>,
```

Comment:
These should not be members of the `Assembler` as it will create issues when we start parsing modules. Instead, we could put them into `AssemblyContext`.

Reply:
Moved to context